### PR TITLE
[release-4.10] Bug 2104889: Fix topology sidebar update issues

### PR DESCRIPTION
**Pull request by jerolimov**
This is a manual backport of #11420 to fix different topology sidebar issues. Incl. also #11683 

See #11420 for more information and a list of all bugs that are fixed and should be doubled checked.

This fixes also the latest sidebar issue [Bug 2103031](https://bugzilla.redhat.com/show_bug.cgi?id=2103031).

### CODE REVIEW

File: frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx [hunk 1]
```diff
@@ -10,26 +11,29 @@ import { isDynamicEventResourceKind } from '../../utils/fetch-dynamic-eventsourc
...
+) => {
   const resource = getResource(element);
   const data = element.getData();
   if (!resource) {
-    return undefined;
+    return [undefined, true, undefined];
   }
   if (
     isDynamicEventResourceKind(referenceFor(resource)) ||
     (resource.kind === CamelKameletBindingModel.kind && data.kameletType === KameletType.Source)
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx:24]:
```suggestion
    (resource.kind === CamelKameletBindingModel.kind && data.data.kameletType === KameletType.Source)
```

**Review by vikram-raj — APPROVED** [review]
Verified all the scenarios and it works as expected.
/lgtm

Kapture 2022-07-08 at 21 17 18

**Comment by sanketpathak** [comment]
[Screencast from 07-08-2022 10:00:20 PM.webm](https://user-images.githubusercontent.com/10252230/178035328-e6aadb5c-44d8-4e26-8aee-9b8747a74002.webm)
/label cherry-pick-approved

**Pull request merged**